### ISSUE TITLE
✅ 🐞 Image Carousel No Longer Showing Caption

### ISSUE THREAD

**Issue by Kubot12** [elementor/elementor#28946]
### Prerequisites

- [X] I have searched for similar issues in open and closed tickets and cannot find a duplicate.
- [X] I have troubleshooted my issue, and it still exists against the latest stable version of Elementor.

### Description

After the latest Elementor plugin update, figcaption in the image carousel stopped showing _only on mobile_. 

What works:
figcaption works on desktop even when the browser window is resized.
figcaption works on the preview tool in wp-admin even in mobile size.
The image and scrolling of the image works fine on mobile.

What is broken:
figcaption does not show in Safari or Chrome browsers in normal and incognito modes on two different model iPhones.

Troubleshooting:
This is happening on two different websites and on and two different pages. Have tried clearing phone and website cache. Have tried the tools > regenerate in Elementor. There are no settings on image carousel to turn off the captions only on mobile. I turned off lazyload. I removed all custom CSS. I tried all kinds of CSS (ie. display: block !important;), but it seems to be something specific to mobile devices going wrong. 

### Steps to reproduce

1. Visit this test site homepage on mobile and look for the image scroll: https://wordpress-1319625-4822077.cloudwaysapps.com/
2. You will see a caption on desktop, but not mobile

(The custom CSS is added back in, but the caption still doesn't show even if this is removed).

### Expected behavior

The caption should show on mobile and desktop.

### Isolating the problem

- [ ] This bug happens when only the Elementor (and Elementor Pro) plugins are active.
- [X] This bug happens with the Hello Elementor theme active.
- [X] I can reproduce this bug consistently by following the steps I described above.

### Elementor System Info

<details>
<summary>Click to reveal</summary>

````txt


== Server Environment ==
	Operating System: Linux
	Software: Apache/2.4.62 (Debian)
	MySQL version: mariadb.org binary distribution v10.6.19
	PHP Version: 8.1.30
	PHP Memory Limit: 512M
	PHP Max Input Vars: 2500
	PHP Max Post Size: 50M
	GD Installed: Yes
	ZIP Installed: Yes
	Write Permissions: All right
	Elementor Library: Connected

== WordPress Environment ==
	Version: 6.6.2
	Site URL: https://wordpress-1319625-4822077.cloudwaysapps.com
	Home URL: https://wordpress-1319625-4822077.cloudwaysapps.com
	WP Multisite: No
	Max Upload Size: 50 MB
	Memory limit: 40M
	Max Memory limit: 512M
	Permalink Structure: /blog/%postname%/
	Language: en_US
	Timezone: 0
	Debug Mode: Inactive

== Theme ==
	Name: Hello Elementor
	Version: 3.1.1
	Author: Elementor Team
	Child Theme: No

== User ==
	Role: administrator
	WP Profile lang: en-US
	User Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/129.0.0.0 Safari/537.36

== Active Plugins ==
	Advanced Custom Fields
		Version: 6.3.6.1
		Author: WP Engine

	Breeze
		Version: 2.1.15
		Author: Cloudways

	Elementor
		Version: 3.24.6
		Author: Elementor.com

	Elementor Pro
		Version: 3.23.3
		Author: Elementor.com

	Image Optimizer by Elementor – Compress, Resize and Optimize Images
		Version: 1.5.2
		Author: Elementor.com

	MalCare WordPress Security Plugin - Malware Scanner, Cleaner, Security Firewall
		Version: 5.77
		Author: MalCare Security

	Object Cache Pro
		Version: 1.21.2
		Author: Rhubarb Group

	Prevent Browser Caching
		Version: 2.3.5
		Author: Kostya Tereshchuk

	Rank Math SEO
		Version: 1.0.229
		Author: Rank Math

	Rank Math SEO PRO
		Version: 3.0.71
		Author: Rank Math

	WP File Manager
		Version: 8.0
		Author: mndpsingh287


== Must-Use Plugins ==
	Asset CleanUp Pro: Plugin Filtering
		Version: 1.0
		Author: Gabriel Livan


== Elements Usage ==
	
	archive : 2
		button : 2
		container : 5
		divider : 1
		heading : 7
		icon : 3
		icon-list : 1
		image : 1
		image-box : 7
		post-info : 1
		posts : 1
		social-icons : 1
		text-editor : 6
	footer : 1
		icon-list : 1
		text-editor : 1
	header : 15
		button : 2
		image : 1
		nav-menu : 1
	page : 7
		button : 2
		call-to-action : 24
		container : 14
		divider : 19
		google_maps : 1
		heading : 73
		icon-box : 3
		image : 24
		social-icons : 1
		spacer : 20
		star-rating : 15
		text-editor : 74
		video-playlist : 1
	popup : 4
		button : 1
		heading : 1
		html : 1
	section : 6
		button : 2
		container : 20
		divider : 14
		heading : 62
		icon-box : 1
		icon-list : 2
		image : 22
		social-icons : 1
		spacer : 8
		star-rating : 18
		text-editor : 38
	widget : 0
		blockquote : 1
		button : 6
		divider : 1
		heading : 1
		icon : 1
		icon-box : 1
		icon-list : 2
		image-carousel : 1
		nav-menu : 3
		posts : 1
		table-of-contents : 1
		testimonial-carousel : 1
		toggle : 2
	wp-post : 10
		container : 20
		divider : 50
		heading : 183
		icon : 2
		icon-box : 8
		icon-list : 69
		image : 25
		image-box : 3
		price-table : 5
		table-of-contents : 7
		text-editor : 197
		toggle : 1
		video : 2
	wp-page : 10
		accordion : 1
		author-box : 1
		button : 7
		container : 6
		counter : 1
		divider : 3
		google_maps : 1
		heading : 49
		html : 1
		icon-box : 12
		icon-list : 10
		image : 23
		image-box : 13
		image-carousel : 3
		image-gallery : 1
		post-info : 2
		posts : 2
		price-table : 14
		spacer : 12
		star-rating : 2
		tabs : 1
		text-editor : 42
		theme-post-title : 1
	single-post : 1
		author-box : 1
		button : 2
		divider : 1
		heading : 3
		html : 1
		post-info : 2
		posts : 1
		spacer : 1
		text-editor : 1
		theme-post-content : 1
		theme-post-title : 1
		toggle : 1
	error-404 : 1
		button : 1
		heading : 3
		text-editor : 1


== Settings ==
	
	cpt_support: post, page
	disable_color_schemes: yes
	disable_typography_schemes: yes
	allow_tracking: yes
	unfiltered_files_upload: 1
	font_display: swap
	load_fa4_shim: yes
	meta_generator_tag: 1


== Features ==
	Custom Fonts: 0
	Custom Icons: 0

== Integrations ==
	


== Elementor Experiments ==
	Improved CSS Loading: Inactive by default
	Inline Font Icons: Active
	Additional Custom Breakpoints: Inactive
	Flexbox Container: Active
	Grid Container: Inactive
	Upgrade Swiper Library: Inactive
	Nested Elements Performance: Active by default
	Optimized Control Loading: Active
	Optimized Markup: Inactive by default
	Conditionally load Swiper CSS files: Inactive by default
	Plugin Onboarding: Active by default
	Default to New Theme Builder: Inactive
	Header & Footer: Inactive
	Elementor Home Screen: Active by default
	Build with AI: Inactive
	Landing Pages: Inactive
	Launchpad Checklist: Inactive by default
	Nested Elements: Inactive by default
	Editor Top Bar: Inactive by default
	Element Caching: Inactive
	Link In Bio: Active by default
	Floating Buttons: Active by default
	Elementor Editor Events: Inactive by default
	Atomic Widgets: Inactive by default
	Form Submissions: Inactive
	Menu: Inactive by default
	Taxonomy Filter: Active by default
	Search: Inactive by default


== Log ==
	
JS: showing 20 of 22JS: 2024-09-15 21:27:44 [error X 12][../wp-includes/js/jquery/jquery.min.js?ver=3.7.1.1726435657:2:37833] Cannot create property 'guid' on string 'c102widgetsticky' 
JS: 2024-09-15 21:27:44 [error X 6][../wp-includes/js/jquery/jquery.min.js?ver=3.7.1.1726435657:2:37833] Cannot create property 'guid' on string 'c105widgetsticky' 
JS: 2024-09-15 21:27:44 [error X 8][../wp-includes/js/jquery/jquery.min.js?ver=3.7.1.1726435657:2:37833] Cannot create property 'guid' on string 'c108widgetsticky' 
JS: 2024-09-15 21:27:49 [error X 7][../wp-includes/js/jquery/jquery.min.js?ver=3.7.1.1726435657:2:37833] Cannot create property 'guid' on string 'c36widgetsticky' 
JS: 2024-09-15 21:31:47 [error X 9][../wp-includes/js/jquery/jquery.min.js?ver=3.7.1.1726435902:2:37833] Cannot create property 'guid' on string 'c88widgetsticky' 
JS: 2024-09-20 18:11:51 [error X 2][../wp-content/plugins/elementor/assets/js/editor.min.js?ver=3.24.3:3:1272139] Cannot read properties of null (reading 'getBoundingClientRect') 
JS: 2024-09-24 00:59:26 [error X 1][../wp-content/plugins/elementor/assets/js/editor.min.js?ver=3.24.4:3:664925] Cannot read properties of undefined (reading 'isDesignable') 
JS: 2024-09-24 22:49:43 [error X 15][../wp-content/plugins/elementor/assets/js/common-modules.min.js?ver=3.24.4:2:56144] Model.isValidChild() should be implemented, please provide 'isValidChild' functionality. 
JS: 2024-09-26 14:29:36 [error X 1][../wp-content/plugins/elementor/assets/js/editor.min.js?ver=3.24.4:3:836902] Cannot read properties of undefined (reading 'toLowerCase') 
JS: 2024-09-29 13:17:32 [error X 3][../wp-includes/js/jquery/jquery.min.js?ver=3.7.1.1727615845:2:37833] Cannot create property 'guid' on string 'c67sectionsticky' 
JS: 2024-09-29 13:17:32 [error X 1][../wp-includes/js/jquery/jquery.min.js?ver=3.7.1.1727615845:2:37833] Cannot create property 'guid' on string 'c100sectionsticky' 
JS: 2024-09-29 13:17:32 [error X 1][../wp-includes/js/jquery/jquery.min.js?ver=3.7.1.1727615845:2:37833] Cannot create property 'guid' on string 'c98sectionsticky' 
JS: 2024-09-29 13:17:32 [error X 4][../wp-includes/js/jquery/jquery.min.js?ver=3.7.1.1727615845:2:37833] Cannot create property 'guid' on string 'c78widgetsticky' 
JS: 2024-09-29 13:17:32 [error X 2][../wp-includes/js/jquery/jquery.min.js?ver=3.7.1.1727615845:2:37833] Cannot create property 'guid' on string 'c111widgetsticky' 
JS: 2024-09-29 13:17:33 [error X 1][../wp-includes/js/jquery/jquery.min.js?ver=3.7.1.1727615845:2:37833] Cannot create property 'guid' on string 'c91widgetsticky' 
JS: 2024-09-30 16:13:03 [error X 2][../wp-includes/js/jquery/jquery.min.js?ver=3.7.1.1727712769:2:37833] Cannot create property 'guid' on string 'c97sectionsticky' 
JS: 2024-09-30 16:13:03 [error X 2][../wp-includes/js/jquery/jquery.min.js?ver=3.7.1.1727712769:2:37833] Cannot create property 'guid' on string 'c95sectionsticky' 
JS: 2024-09-30 16:13:03 [error X 2][../wp-includes/js/jquery/jquery.min.js?ver=3.7.1.1727712769:2:37833] Cannot create property 'guid' on string 'c99widgetsticky' 
JS: 2024-09-30 16:13:20 [error X 1][../wp-includes/js/jquery/jquery.min.js?ver=3.7.1.1727712769:2:37833] Cannot create property 'guid' on string 'c43sectionsticky' 
JS: 2024-10-11 17:43:37 [error X 1][../wp-content/plugins/elementor/assets/js/frontend.min.js?ver=3.24.6.1728668613:2:20010] Cannot read properties of undefined (reading 'top') 

Log: showing 17 of 172024-09-16 18:06:02 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor',
  'from' => '3.24.0',
  'to' => '3.24.2',
)]
2024-09-16 18:06:07 [info] elementor::elementor_updater Started 
2024-09-16 18:06:07 [info] Elementor/Upgrades - _on_each_version Start  
2024-09-16 18:06:07 [info] Elementor/Upgrades - _on_each_version Finished 
2024-09-16 18:06:08 [info] Elementor/Upgrades - _on_each_version Start  
2024-09-16 18:06:08 [info] Elementor/Upgrades - _on_each_version Finished 
2024-09-16 18:06:08 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.24.0',
  'to' => '3.24.2',
)]
2024-09-16 18:06:10 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor',
  'from' => '3.24.0',
  'to' => '3.24.2',
)]
2024-10-03 11:42:28 [info] elementor::elementor_updater Started 
2024-10-03 11:42:28 [info] Elementor/Upgrades - _on_each_version Start  
2024-10-03 11:42:32 [info] Elementor/Upgrades - _on_each_version Finished 
2024-10-03 11:42:32 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.24.2',
  'to' => '3.24.5',
)]
2024-10-11 11:28:55 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor',
  'from' => '3.24.5',
  'to' => '3.24.6',
)]
2024-10-11 11:29:09 [info] elementor::elementor_updater Started 
2024-10-11 11:29:09 [info] Elementor/Upgrades - _on_each_version Start  
2024-10-11 11:29:19 [info] Elementor/Upgrades - _on_each_version Finished 
2024-10-11 11:29:19 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.24.5',
  'to' => '3.24.6',
)]

PHP: showing 4 of 4PHP: 2024-09-16 18:06:07 [warning X 1][../wp-content/plugins/elementor/core/files/manager.php::113] unlink(../wp-content/uploads/elementor/css/post-3132.css): No such file or directory [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2024-09-20 16:16:57 [notice X 5][../wp-content/plugins/elementor/core/files/file-types/svg.php::165] explode(): Passing null to parameter #2 ($string) of type string is deprecated [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2024-09-20 17:56:22 [warning X 18][../wp-content/plugins/elementor/includes/managers/image.php::113] Trying to access array offset on value of type bool [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2024-09-20 18:16:49 [warning X 1][../wp-content/plugins/elementor-pro/modules/query-control/module.php::387] Undefined array key "display" [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]



== Elementor - Compatibility Tag ==
	
	Elementor Pro: Incompatible
	Image Optimizer by Elementor – Compress, Resize and Optimize Images: Compatibility not specified

== Elementor Pro - Compatibility Tag ==

````

</details>

### Agreement

- [X] I agree that my issue may be closed without action if it doesn't meet all the requirements.

**Comment by lucaspanwang** [comment]
Hi, I also met the same issue. But for my website, the captions can not be showing on both desktop and mobile. And I looked at your website, https://wordpress-1319625-4822077.cloudwaysapps.com/, there is also no captions on desktop.
So I don't know what you mean about it can be showing on desktop.
<img width="1359" alt="Screenshot 2024-10-14 at 1 25 41 PM" src="https://github.com/user-attachments/assets/c93af2a5-9bf9-4a46-b8e5-f1e30025162f">
<img width="646" alt="Screenshot 2024-10-14 at 1 25 51 PM" src="https://github.com/user-attachments/assets/15968e08-655e-4912-9307-f932b4bd6bb6">

**Comment by Kubot12** [comment]
Great catch! I CAN see the captions logged in on desktop, even after clearing cache or using incognito. Interesting.

But, I cannot see on desktop or mobile when not logged in.

**Comment by attilaparcsami** [comment]
Hey guys. 
I had the same problem on Elementor 3.24.6 in the last few hours. The caption was ok on the Elementor editor, but not on the live page. I noticed the problem on iPhone with Chrome browser, but it didn't work on the desktop (Brave, Chrome) either. So I found your discussion. Unfortunately I didn't found the answer, so I tried to roll back to a previous version of Elementor.
After a google search, I installed the WP Rollback (by WP Rollback) plugin. Than I Installed a WP migration tool (Matt - WPress Doctor has fckn amazing videos) and I made an export of my entire site to have a backup file, before the rollback is kill everything. Than, after I deleted the migration tool (with an interesting feeling in my back) I made a last check on my site, and it's working :D 

I did nothing with the Elementor. I just installed and uninstalled those two plugins and its working.
I know, it's not a bug fix, but maybe it will be working for you guys.
https://www.fl-kurzus.hu/

**Comment by lucaspanwang** [comment]
> Hey guys. I had the same problem on Elementor 3.24.6 in the last few hours. The caption was ok on the Elementor editor, but not on the live page. I noticed the problem on iPhone with Chrome browser, but it didn't work on the desktop (Brave, Chrome) either. So I found your discussion. Unfortunately I didn't found the answer, so I tried to roll back to a previous version of Elementor. After a google search, I installed the WP Rollback (by WP Rollback) plugin. Than I Installed a WP migration tool (Matt - WPress Doctor has fckn amazing videos) and I made an export of my entire site to have a backup file, before the rollback is kill everything. Than, after I deleted the migration tool (with an interesting feeling in my back) I made a last check on my site, and it's working :D
> 
> I did nothing with the Elementor. I just installed and uninstalled those two plugins and its working. I know, it's not a bug fix, but maybe it will be working for you guys. https://www.fl-kurzus.hu/

Thank you for sharing! It's an amazing strategy. It also works to my website.

**Comment by Zabien** [comment]
+1 Having the same issue (and hope for a solution without a rollback).

**Comment by BouquetAntoine** [comment]
+1 
caused by Utils::has_invalid_post_permissions($attachment_post) inside get_image_caption, which prevents caption to be fetch for non admin users (attachment don't have post_status=publish, so it returns false and prevent caption to be displayed)

Using this fix until resolved : 
`function custom_allow_media_edit_permission($allcaps, $cap, $args)
{
    if (isset($cap[0]) && $cap[0] === 'edit_others_posts') {
        if (!is_user_logged_in()) {
            $post_id = isset($args[2]) ? $args[2] : 0;
            if ($post_id) {
                $post = get_post($post_id);
                if ($post && $post->post_type === 'attachment' && $post->post_status === 'inherit') {
                    $allcaps['edit_others_posts'] = true;
                }
            }
        }
    }
    return $allcaps;
}

add_filter('user_has_cap', 'custom_allow_media_edit_permission', 10, 3);
`
It allows non-logged users to access "protected" attachment so it may be dangerous

**Comment by BirdhouseMN** [comment]
This issue appears to be with Elementor versions **3.24.7** and **3.24.6**.

Once I rolled back to Version **3.24.5**, my captions started to display correctly again on my service tiles (reference link: https://goodhustlemedia.com/services).

I hope this is fixed soon, I would like to run the latest version of Elementor.

**Comment by ryanwdesigner** [comment]
Having the same issue, caption not visible on desktop or mobile.

**Comment by nicholaszein** [comment]
Hey there!

We have great news! 🙌 

### 📢 We're happy to announce that this issue was solved with the release of Elementor Core v3.25.2! 🥳 

✅ Feel free to check it out and update your plugin to the new version!

> _Check the changelog for more information:_
> [Changelog of Elementor and Elementor Pro](https://elementor.com/help/changelog-of-elementor/)

Cheers 🥂
